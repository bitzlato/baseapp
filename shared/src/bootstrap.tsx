import { StrictMode } from 'react';
import ReactDOM from 'react-dom';
import { Main } from './Main';

ReactDOM.render(
  <StrictMode>
    <Main />
  </StrictMode>,
  document.getElementById('root'),
);
