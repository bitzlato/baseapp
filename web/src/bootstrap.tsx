/* eslint-disable global-require */
import 'bootstrap/dist/css/bootstrap-grid.min.css';
import 'bootstrap/dist/css/bootstrap.min.css';
import * as ReactDOM from 'react-dom';
import 'react-perfect-scrollbar/dist/css/styles.css';
import { Provider } from 'react-redux';

import { App } from './App';
import { AppContextProvider } from './components/app/AppContextProvider';
import './index.pcss';
import { rootSaga } from './modules';
import { rangerSagas } from './modules/public/ranger';
import { rangerMiddleware, sagaMiddleware, store } from './store';

if (!Intl?.PluralRules) {
  require('@formatjs/intl-pluralrules/polyfill');
  require('@formatjs/intl-pluralrules/locale-data/en');
  require('@formatjs/intl-pluralrules/locale-data/ru');
}

// eslint-disable-next-line @typescript-eslint/ban-ts-comment
// @ts-ignore
if (!Intl?.RelativeTimeFormat) {
  require('@formatjs/intl-relativetimeformat/polyfill');
  require('@formatjs/intl-relativetimeformat/locale-data/en');
  require('@formatjs/intl-relativetimeformat/locale-data/ru');
}

sagaMiddleware.run(rootSaga);
rangerMiddleware.run(rangerSagas);

ReactDOM.render(
  <Provider store={store}>
    <AppContextProvider>
      <App />
    </AppContextProvider>
  </Provider>,
  document.getElementById('root') as HTMLElement,
);
