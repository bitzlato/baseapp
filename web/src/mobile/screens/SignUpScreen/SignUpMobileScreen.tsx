import { FC } from 'react';
import { Box } from 'web/src/components/Box/Box';
import { SignUp } from 'web/src/containers/SignUp/SignUp';

export const SignUpMobileScreen: FC = () => {
  return (
    <Box col spacing="sm">
      <div />
      <Box bgColor="body">
        <SignUp />
      </Box>
    </Box>
  );
};
