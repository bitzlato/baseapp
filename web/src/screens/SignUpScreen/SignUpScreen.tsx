import { FC } from 'react';
import { Container } from 'web/src/components/ui/Container';
import { Card } from 'web/src/components/Card/Card';
import { SignUp } from 'web/src/containers/SignUp/SignUp';

export const SignUpScreen: FC = () => {
  return (
    <Container maxWidth="sm" my="6x">
      <Card>
        <SignUp />
      </Card>
    </Container>
  );
};
