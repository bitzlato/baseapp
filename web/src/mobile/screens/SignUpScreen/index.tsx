import * as React from 'react';
import { useIntl } from 'react-intl';
import { useHistory } from 'react-router';
import { SignUpScreen } from '../../../screens/SignUpScreen';
import { MobileModal } from '../../components';

const SignUpMobileScreen: React.FC = () => {
  const history = useHistory();
  const intl = useIntl();

  return (
    <div>
      <MobileModal
        isOpen
        onClose={() => history.push('/trading')}
        onBack={() => history.push('/signinb')}
        backTitle={intl.formatMessage({ id: 'page.body.landing.header.button2' })}
        title={intl.formatMessage({ id: 'page.body.landing.header.button3' })}
      >
        <SignUpScreen />
      </MobileModal>
    </div>
  );
};

export { SignUpMobileScreen };
