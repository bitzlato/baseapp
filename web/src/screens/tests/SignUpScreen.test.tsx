import { shallow } from 'enzyme';
import React from 'react';
import { TestComponentWrapper } from 'src/lib/test';
import { IntlProps } from '../../bootstrap';
import { SignUpScreen } from '../SignUpScreen';

const setup = (props: Partial<IntlProps> = {}) =>
  shallow(
    <TestComponentWrapper>
      <SignUpScreen />
    </TestComponentWrapper>,
  );

describe('SignUpScreen', () => {
  it('should render', () => {
    const wrapper = setup().render();
    expect(wrapper).toMatchSnapshot();
  });
});
