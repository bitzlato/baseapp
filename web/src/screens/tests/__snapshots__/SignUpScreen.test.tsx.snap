// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignUpScreen should render 1`] = `
<div
  class="pg-sign-up-screen"
>
  <div
    class="pg-sign-up-screen__container"
  >
    <form>
      <div
        class="cr-sign-up-form"
      >
        <div
          class="cr-sign-up-form__options-group"
        >
          <div
            class="cr-sign-up-form__option"
          >
            <div
              class="cr-sign-up-form__option-inner cr-sign-in-form__tab-signin"
            >
              Sign In
            </div>
          </div>
          <div
            class="cr-sign-up-form__option"
          >
            <div
              class="cr-sign-up-form__option-inner __selected"
            >
              Sign Up
            </div>
          </div>
        </div>
        <div
          class="cr-sign-up-form__form-content"
        >
          <div
            class="cr-sign-up-form__group"
          >
            <div
              class="custom-input"
            >
              <label
                class="cr-sign-up-form__label"
              />
              <input
                autofocus=""
                class="form-control"
                placeholder="Email"
                type="email"
                value=""
              />
            </div>
          </div>
          <div
            class="cr-sign-up-form__group"
          >
            <div
              class="custom-input"
            >
              <label
                class="cr-sign-up-form__label"
              />
              <input
                class="form-control"
                placeholder="Password"
                type="password"
                value=""
              />
            </div>
          </div>
          <div
            class="cr-sign-up-form__group"
          >
            <div
              class="custom-input"
            >
              <label
                class="cr-sign-up-form__label"
              />
              <input
                class="form-control"
                placeholder="Confirm Password"
                type="password"
                value=""
              />
            </div>
          </div>
          <div
            class="custom-control custom-checkbox"
          >
            <input
              class="custom-control-input"
              id="agreeWithTerms"
              type="checkbox"
            />
            <label
              class="custom-control-label"
              for="agreeWithTerms"
              title=""
            >
              I have read and agree to the Terms of Service
            </label>
          </div>
          <div
            class="cr-sign-up-form__button-wrapper"
          >
            <button
              class="btn btn-primary btn-block btn-lg"
              disabled=""
              type="button"
            >
              Sign Up
            </button>
          </div>
        </div>
      </div>
    </form>
  </div>
</div>
`;
